### PR TITLE
RT-434: Fix/activity stream data loss

### PR DESCRIPTION
**Pull request by NickFitz**
Some models have many instances with identical values for `last_modified`. The `activity_stream` pagination depends on ordering by that field. As the instances are unordered within the ordering by `last_modified`, it becomes possible for any given instance to appear on multiple pages, or not to appear on any page.

This can be addressed by adding `id` as a second-level ordering.

### CODE REVIEW

File: update_supply_chain_information/activity_stream/test/test_pagination.py [hunk 1]
```diff
@@ -154,3 +156,91 @@ def test_paginated_response_has_no_next_link_on_empty_page(
...
+                page_length * 2, supply_chain=supply_chain
+            )
+            all_last_modified = [
+                strategic_action.last_modified
+                for strategic_action in StrategicAction.objects.all()
+            ]
+            all_last_modified_set = {
+                strategic_action.last_modified
+                for strategic_action in StrategicAction.objects.all()
+            }
```
Comment (dev-digital) [inline update_supply_chain_information/activity_stream/test/test_pagination.py:191]:
```
set(all_last_modified)
```
is more explicit

File: update_supply_chain_information/activity_stream/test/test_pagination.py [hunk 2]
```diff
@@ -154,3 +156,91 @@ def test_paginated_response_has_no_next_link_on_empty_page(
...
+            page_item_ids = {item["json"]["id"] for item in first_page_items}
+            next_link = pagination.get_next_link()
+            while next_link:
+                request = rf.get(next_link)
+                drf_request = Request(request)
+                next_page_items = pagination.paginate_queryset(queryset, drf_request)
+                next_page_item_ids = {item["json"]["id"] for item in next_page_items}
+                page_item_ids = page_item_ids | next_page_item_ids
+                next_link = pagination.get_next_link()
+            assert all_item_ids.issubset(page_item_ids)
```
Comment (dev-digital) [inline update_supply_chain_information/activity_stream/test/test_pagination.py:246]: Minor:

You are better off going for equality here as all ids are expected to be present in *page_item_ids* if I read the script correctly.

**Review by dev-digital — APPROVED** [review]
Minor comments.

I presume you want to deploy this to master which would be another PR ?

File: update_supply_chain_information/activity_stream/test/test_pagination.py (hunk 2 above)
Comment (NickFitz) [reply update_supply_chain_information/activity_stream/test/test_pagination.py:246]: Not quite: one or more of the pages will also include the IDs of the default GovDepartment and of the SupplyChain fixture (and I think a User too). So the `page_item_ids` set will end up with a small number of IDs that aren't in the StrategicAction IDs set.

**Pull request merged**